mesonbuild/cmake/client.py

```diff
@@ -62,21 +64,21 @@ def log(self) -> None:
 class RequestBase(MessageBase):
     cookie_counter = 0
 
-    def __init__(self, msg_type: str):
+    def __init__(self, msg_type: str) -> None:
```

Comment:
we should really decide what we want to do with `__init__`, mypy assumes that the return type is `None`, the only case you need the return annotation is for cases like `__init__(self) -> None`, where there are no arguments, and thus mypy can't infer that the method is supposed to be typed.

Reply:
I would vote for always use `-> None` because it is required for all other functions with no return type and it is also sometimes required for `__init__` itself. So the most consistent thing we could do would be to always use `-> None` IMHO.